chain-signatures/node/src/protocol/presignature.rs

```diff
@@ -403,48 +565,15 @@ impl PresignatureManager {
         }
     }
 
-    pub fn take_mine(&mut self) -> Option<Presignature> {
-        let my_presignature_id = self.mine.pop_front()?;
-        tracing::info!(my_presignature_id, "take presignature of mine");
-        // SAFETY: This unwrap is safe because taking mine will always succeed since it is only
-        // present when generation completes where the determination of ownership is made.
-        Some(self.take(my_presignature_id).unwrap())
-    }
-
-    pub fn take(&mut self, id: PresignatureId) -> Result<Presignature, GenerationError> {
-        if let Some(presignature) = self.presignatures.remove(&id) {
-            self.gc.insert(id, Instant::now());
-            tracing::info!(id, "took presignature");
-            return Ok(presignature);
-        }
-
-        if self.generators.contains_key(&id) {
-            tracing::warn!(id, "presignature is still generating");
-            return Err(GenerationError::PresignatureIsGenerating(id));
-        }
-        if self.gc.contains_key(&id) {
-            tracing::warn!(id, "presignature was garbage collected");
-            return Err(GenerationError::PresignatureIsGarbageCollected(id));
-        }
-        tracing::warn!(id, "presignature is missing");
-        Err(GenerationError::PresignatureIsMissing(id))
-    }
-
-    pub fn insert_mine(&mut self, presig: Presignature) {
-        tracing::debug!(id = ?presig.id, "inserting presignature");
-        // Remove from taken list if it was there
-        self.gc.remove(&presig.id);
-        self.mine.push_back(presig.id);
-        self.presignatures.insert(presig.id, presig);
-    }
-
     /// Pokes all of the ongoing generation protocols and returns a vector of
     /// messages to be sent to the respective participant.
     ///
     /// An empty vector means we cannot progress until we receive a new message.
-    pub fn poke(&mut self) -> Vec<(Participant, PresignatureMessage)> {
+    pub async fn poke(&mut self) -> Vec<(Participant, PresignatureMessage)> {
         let mut messages = Vec::new();
         let mut errors = Vec::new();
+        let mut new_presignatures = Vec::new();
+        let mut new_mine_presignatures = Vec::new();
```

Comment:
now that I see this, it's better for us to just have an extra bool field denoting the ownership of the presignature even though we already maintain a list of mine presignatures in redis. Would simplify the code a bit

Reply:
Redis is not designed for such data structures. For example, it will need to deserialize all triples to count mine.